### PR TITLE
Fail early if not enough ground truth data provided, fix #42

### LINKED ISSUE #42: Add exception handling for GT sets which consist of too few files
Cf. https://github.com/OCR-D/ocrd-train/issues/40

Having fewer than 10 lines results in an empty eval list.

### PR DESCRIPTION
**Pull request by kba**

### CODE REVIEW

**Comment by wrznr** [comment]
```
Fail if 1 - RATIO * no < 1
```

**Comment by kba** [comment]
> Fail if (1 - RATIO) * no < 1

Implemented in db1669a

**Comment by wrznr** [comment]
@kba So, we are ready to merge here?

File: README.md [hunk 1]
```diff
@@ -99,6 +99,7 @@ Run `make help` to see all the possible targets and variables:
     PSM                Page segmentation mode. Default: 6
     RANDOM_SEED        Random seed for shuffling of the training data. Default: 0
     RATIO_TRAIN        Ratio of train / eval training data. Default: 0.90
+    MIN_GT_FILES       Minimum number of GT files. Default: 10
 ```
```
Comment (stweil) [inline README.md]: This looks like a relict from the original pull request.

File: Makefile [hunk 2]
```diff
@@ -1,6 +1,6 @@
 export
 
-# Make sure that sort always uses the same sort order.
+## Make sure that sort always uses the same sort order.
 LC_ALL := C
```
Comment (stweil) [inline Makefile]: I am curious: what do the double `##` signify?

**Comment by stweil** [comment]
I'm afraid my latest commits have created a merge conflict.

**Comment by stweil** [comment]
I just tested the scenario with a very small set of ground truth and came to the conclusion that there is no need to check the size of the set.

We currently have a rounding problem: a set of 9999 results is split in 8999 and 999, so one set member gets lost.

**Comment by wrznr** [comment]
@stweil This PR is a solution for #42 which was triggered by #40. So it seems to be real problem. Do you think we should ignore such edge cases?

**Comment by stweil** [comment]
We could ignore them in the `Makefile` and let the Tesseract code handle them.

Pull request #84 is a new implementation which supersedes #44. It fixes the rounding problem, and I also implemented the corner cases of empty evaluation set and empty training set.

File: README.md (hunk 1 above)
Comment (stweil) [reply README.md]: @kba, this commit is good but conflicts with my change which only fixed `TESSERACT_VERSION`.

File: Makefile (hunk 2 above)
Comment (kba) [reply Makefile]: [`makefile-parser`](https://github.com/kba/makefile-parser) won't pick it up as a comment on that variable and it won't be listed in the [help target](https://github.com/kba/makefile-parser#make-help).

**Comment by stweil** [comment]
@kba, I added your commit for `README.md` to Git master, so I think this pull request can now be closed.

**Comment by stweil** [comment]
@kba, I think we can delete the branch.

**Comment by kba** [comment]
Sure go ahead

**Pull request closed without merging**